discord/automod.py

```diff
@@ -0,0 +1,420 @@
+        The presets used with the preset keyword filter.
+
+    .. versionadded:: 2.0
+    """
+
+    __slots__ = ('type', 'keyword_filter', 'presets')
+
+    def __init__(
+        self,
+        *,
+        type: AutoModRuleTriggerType,
+        keyword_filter: Optional[List[str]] = None,
+        presets: Optional[AutoModPresets] = None,
+    ) -> None:
+        self.type: AutoModRuleTriggerType = type
+        self.keyword_filter: Optional[List[str]] = keyword_filter
+        self.presets: Optional[AutoModPresets] = presets
+
+        if self.type is AutoModRuleTriggerType.keyword and self.keyword_filter is None:
+            raise ValueError('keyword_filter must be set if type is keyword.')
+        elif self.type is AutoModRuleTriggerType.keyword_preset and self.presets is None:
+            raise ValueError('presets must be set if type is keyword_preset.')
+
+    def __repr__(self) -> str:
+        return f"<AutoModTrigger type={self.type}>"
+
+    @classmethod
+    def from_data(cls: Type[Self], type: int, data: Optional[AutoModerationTriggerMetadataPayload]) -> Self:
+        type_ = try_enum(AutoModRuleTriggerType, type)
+        if type_ is AutoModRuleTriggerType.keyword:
+            return cls(type=type_, keyword_filter=data['keyword_filter'])  # type: ignore # unable to typeguard due to outer payload
+        elif type_ is AutoModRuleTriggerType.keyword_preset:
+            return cls(type=type_, presets=AutoModPresets._from_value(data['presets']))  # type: ignore # unable to typeguard due to outer payload
+        return cls(type=type_)
+
+    def to_metadata_dict(self) -> AutoModerationTriggerMetadataPayload:
+        if self.type is AutoModRuleTriggerType.keyword:
+            return {'keyword_filter': self.keyword_filter}  # type: ignore # guarded by the type check
+        elif self.type is AutoModRuleTriggerType.keyword_preset:
+            return {'presets': self.presets.value}  # type: ignore # guarded by the type check
+
+        return {}
+
+
+class AutoModRule:
+    """Represents an auto moderation rule.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The ID of the rule.
+    guild: :class:`Guild`
+        The guild the rule is for.
+    name: :class:`str`
+        The name of the rule.
+    creator_id: :class:`int`
+        The ID of the user that created the rule.
+    trigger: :class:`AutoModTrigger`
+        The rule's trigger.
+    enabled: :class:`bool`
+        Whether the rule is enabled.
+    raw_exempt_roles: List[:class:`int`]
+        The IDs of the roles that are exempt from the rule.
+    raw_exempt_channels: List[:class:`int`]
+        The IDs of the channels that are exempt from the rule.
+
+    .. versionadded:: 2.0
+    """
+
+    __slots__ = (
+        '_state',
+        'id',
+        'guild',
+        'name',
+        'creator_id',
+        'event_type',
+        'trigger',
+        'enabled',
+        'raw_exempt_roles',
+        'raw_exempt_channels',
```

Comment:
```suggestion
        'exempt_role_ids',
        'exempt_channel_ids',
```

I know this was done for consistency with `Message.raw_` methods but the context is different.